### PR TITLE
AG-9212 - Conflate column and bar series types

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-9212

Focuses on updating the implementation, tests and examples - documentation will need another pass.

### CODE REVIEW

**Review by lsjroberts — APPROVED** [review]
LGTM. Only thought is should we rename `range-bar` to `range` to match the change from `waterfall-bar` to `waterfall`?

File: packages/ag-charts-community/src/chart/series/cartesian/rangeBarOptions.ts [hunk 1]
```diff
@@ -75,7 +75,9 @@ export type AgRangeBarSeriesLabelPlacement = 'inside' | 'outside';
 /** Configuration for RangeBar series. */
 export interface AgRangeBarSeriesOptions<DatumType = any> extends AgBaseSeriesOptions<DatumType> {
     /** Configuration for the RangeBar series. */
-    type?: 'range-bar' | 'range-column';
+    type: 'range';
```
Comment (manapeirov) [inline packages/ag-charts-community/src/chart/series/cartesian/rangeBarOptions.ts]: should this be `range-bar` as we have type `range-area`?

File: packages/ag-charts-enterprise/src/rangeBar/rangeBar.ts [hunk 2]
```diff
@@ -129,7 +130,7 @@ class RangeBarSeriesLabel extends _Scene.Label {
 
 export class RangeBarSeries extends _ModuleSupport.CartesianSeries<RangeBarContext, _Scene.Rect> {
     static className = 'RangeBarSeries';
-    static type: 'range-bar' | 'range-column' = 'range-bar' as const;
+    static type: 'range-bar' | 'range-bar' = 'range-bar' as const;
```
Comment (manapeirov) [inline packages/ag-charts-enterprise/src/rangeBar/rangeBar.ts:134]: can delete the union type here

File: packages/ag-charts-enterprise/src/waterfall/waterfallModule.ts [hunk 3]
```diff
@@ -1,29 +1,18 @@
...
-    type: 'series',
-    optionsKey: 'series[]',
-    packageType: 'enterprise',
-    chartTypes: ['cartesian'],
-
-    identifier: 'waterfall-bar',
-    instanceConstructor: WaterfallBarSeries,
+    identifier: 'waterfall',
+    instanceConstructor: WaterfallSeries,
     seriesDefaults: WATERFALL_BAR_DEFAULTS,
```
Comment (manapeirov) [inline packages/ag-charts-enterprise/src/waterfall/waterfallModule.ts]: 'WATERFALL_DEFAULTS'

File: packages/ag-charts-website/src/content/docs/charts-waterfall-series/_examples/basic-waterfall-horizontal/main.ts [hunk 4]
```diff
@@ -12,7 +12,8 @@ const options: AgChartOptions = {
     },
     series: [
         {
-            type: 'waterfall-column',
```
Comment (manapeirov) [inline packages/ag-charts-website/src/content/docs/charts-waterfall-series/_examples/basic-waterfall-horizontal/main.ts:15]: Was this example meant to be a vertical column? now it's a  waterfall-bar

File: packages/ag-charts-website/src/content/docs/charts-waterfall-series/_examples/basic-waterfall/main.ts [hunk 5]
```diff
@@ -12,7 +12,7 @@ const options: AgChartOptions = {
     },
     series: [
         {
-            type: 'waterfall-bar',
```
Comment (manapeirov) [inline packages/ag-charts-website/src/content/docs/charts-waterfall-series/_examples/basic-waterfall/main.ts:15]: This one needs direction `horizontal`

**Review by manapeirov — APPROVED** [review]

File: packages/ag-charts-website/src/content/docs/charts-waterfall-series/_examples/basic-waterfall-horizontal/main.ts (hunk 4 above)
Comment (alantreadway) [reply packages/ag-charts-website/src/content/docs/charts-waterfall-series/_examples/basic-waterfall-horizontal/main.ts:15]: I think `git mv` got a bit confused here - the direction configured with the series matches the example name, so I think it's correct,

File: packages/ag-charts-website/src/content/docs/charts-waterfall-series/_examples/basic-waterfall/main.ts (hunk 5 above)
Comment (alantreadway) [reply packages/ag-charts-website/src/content/docs/charts-waterfall-series/_examples/basic-waterfall/main.ts:15]: Ditto

File: packages/ag-charts-community/src/chart/series/cartesian/rangeBarOptions.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/series/cartesian/rangeBarOptions.ts]: Fixed

File: packages/ag-charts-enterprise/src/rangeBar/rangeBar.ts (hunk 2 above)
Comment (alantreadway) [reply packages/ag-charts-enterprise/src/rangeBar/rangeBar.ts:134]: Fixed

File: packages/ag-charts-enterprise/src/waterfall/waterfallModule.ts (hunk 3 above)
Comment (alantreadway) [reply packages/ag-charts-enterprise/src/waterfall/waterfallModule.ts]: Fixed

**Comment by manapeirov** [comment]
LGTM

**Pull request merged**